### ISSUE TITLE
`convert_file` errors with QXcbConnection: Could not connect to display

### ISSUE THREAD

**Issue by DandyDeveloper** [bebraw/pypandoc#144]
I'm not sure whether this dependance is coming from Pandoc directly or wkhtmltopdf. This is a tornado app. 

I have `wkhtmltopdf 0.12.2.4` and `pandoc 1.19.2.1`

This is the convert that is failing: 
```
pypandoc.convert_file(markdown_file, "html5", outputfile=pdf_file, format="markdown_github", extra_args=["-c " + publish_css])
```

Exception:
```
Traceback (most recent call last):
  File "/home/usr/.local/lib/python2.7/site-packages/tornado/web.py", line 1509, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "server.py", line 470, in post
    extra_args=["-c " + publish_css])
  File "/home/usr/.local/lib/python2.7/site-packages/pypandoc/__init__.py", line 140, in convert_file
    outputfile=outputfile, filters=filters)
  File "/home/usr/.local/lib/python2.7/site-packages/pypandoc/__init__.py", line 325, in _convert_input
    'Pandoc died with exitcode "%s" during conversion: %s' % (p.returncode, stderr)
RuntimeError: Pandoc died with exitcode "43" during conversion: QXcbConnection: Could not connect to display
```

I've tried running this with a display running `Xvfb :0 &` but its still failing. Any suggestions? 

This seems more like a Pandoc problem though. Please close if you think this needs to be dealt with by Pandoc.

**Comment by janschulz** [comment]
This is not pypandoc failing, but pandoc itself. The error message sounds like it is something qt related, not sure what in pandoc used QT :-(

My first ideas would be to try setting the DISPLAY env variable to something which has an xserver (e.g. `export DISPLAY=:0` or `os.environ['DISPLAY'] = ':0'`. The second I found in https://github.com/ariya/phantomjs/issues/14376 : `QT_QPA_PLATFORM=offscreen`

**Comment by DandyDeveloper** [comment]
I actually think this is `wkhtmltopdf` 12.4 having the bug still. I'm trying to upgrade to 0.12.4

**Comment by DandyDeveloper** [comment]
I'll leave this here for visibility if anyone has this problem in future. 

You'll need to use the latest binaries from wkhtmltopdf here: https://unix.stackexchange.com/questions/192642/wkhtmltopdf-qxcbconnection-could-not-connect-to-display

I was using version 0.12.2. This is fixed in version 0.12.4.

**Comment by janschulz** [comment]
Thanks for posting the solution!
